### PR TITLE
test: Add complete test suite for `InteractionItem`

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Summary
- Added the tests for the `Interaction Item` component. 
- Added Test IDs to some of the divisions of the base component as there isn't a real accessible (`role` based) query parameter to capture them as they are mostly icons and can be visually distinguished by the users.

## Reviewer guidance
- I have tested all the changes by running the test suite locally

## Testing checklist

- [X] Contributor has fully tested the PR manually
- [X] If there are any front-end changes, before/after screenshots are included
- [X] Critical user journeys are covered by Gherkin stories
- [X] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/views/InteractionList/__tests__/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    routes: new VueRouter(),
+  });
+};
+
+describe('InteractionList', () => {
+  // Check if the selected interaction has the correct cursor
+  // Auto cursor means that the interaction is selected, and pointer cursor means that it is not
+  const isSelected = iteractionItem => iteractionItem.style.cursor === 'auto';
+
+  test('renders correctly and the selected iteraction changes correctly on user clicks', async () => {
```
Comment (AlexVelezLl) [inline kolibri/core/assets/src/views/InteractionList/__tests__/index.spec.js]: Here we are testing several things, we can break up this test, and give a more meaningful name.

1. renders corectly all interactions. (We specifically set the 3 interactions and check the length is correct.
2. just the selectedInteractionIndex is actually selected. (Here we specifically set the `selectedInteractionIndex` prop).
3. renders attempt message correctly.
4. Doesnt emit the selected event when the user clicks on the interaction that is already selected.
5. Emits the selected event when the user clicks on a new interaction item.

Comment (EshaanAgg) [reply kolibri/core/assets/src/views/InteractionList/__tests__/index.spec.js]: Got it. I feel that that the points `2`  and `3` can be clubbed into a single test, and we can wrap all of them in a `describe` block to provide more semantic meaning. I'll update the test suite and request a review when done. Thanks for your time!

**Review by AlexVelezLl — COMMENTED** [review]
Thank you @EshaanAgg! LGTM :tada:.

**Review by MisRob — APPROVED** [review]
Thank you @EshaanAgg and @AlexVelezLl

**Pull request merged**